packages/verified-fetch/src/verified-fetch.ts

```diff
@@ -156,7 +126,8 @@ export class VerifiedFetch {
     this.log.trace('created VerifiedFetch instance')
   }
 
-  private getBlockstore (root: CID, key: string, useSession: boolean, options?: AbortOptions): Blockstore {
+  private getBlockstore (root: CID, resource: string | CID, useSession: boolean, options?: AbortOptions): Blockstore {
+    const key = resourceToSessionCacheKey(resource)
```

Comment:
instead of constructing the key in `fetch` function, construct in here and just use the resource that `FetchHandlerFunctionArg` already has.